accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,632 @@
+
+This has been the subject of some controversy; for instance, in the [proposal to add Sigstore signing to RubyGems](https://github.com/rubygems/rfcs/pull/37), users worried about “vendorization” (making the repository reliant on third parties to function, e.g. GitHub, Google or Microsoft) and privacy (some maintainers are pseudonymous; Sigstore certificates include email addresses by default when signing using the [cosign](https://github.com/sigstore/cosign) CLI tool).
+
+Sigstore supports both email and CI/CD workload-based OIDC identities:
+
+##### Email based identity provider
+For instance, Google, GitHub, Microsoft (supported by Sigstore today) or any other IdP that Sigstore might support.
+
+**Pros**
+- Separate party for trust: it would be difficult for an attacker to compromise the repository and one of these IdPs.
+- Many (but not all!) users have these accounts already.
+- Portable: supports publishing from most environments (requiring a browser and an IdP login).
+- Minimal changes to existing publish workflow for maintainers that are publishing directly.
+- Reliability: these IdPs already work at scale and have full-time support.
+
+**Cons**
+- Privacy: the identities used to sign packages are public (though users can make a pseudonymous account).
+- Privacy: Only a handful of identity providers are supported today that might not suit all users (GitHub, Google or Microsoft).
+- Security: Some users might only have secondary, less secure, accounts on one of the supported services making it an easier target.
+- Does not include provenance information about where the release came from and how it was built.
+- The email might not match the one used on the registry making it hard to validate that it's from a legitimate maintainer.
+
+##### CI/CD workload identity provider
+For instance, Github Actions (supported by Sigstore today) or any other CI/CD system that can provide OIDC workload identities with required claims.
+
+**Pros**
+- Support for pseudonyms/privacy: does not require public identities other than what already corresponds to the package (e.g. repository).
+- Links the package to the source repository and includes information about and how it was built: commit SHA, workflow run id etc.
+- Vendor-neutral: supports any CI/CD system supporting OIDC workload identities.
+- OIDC workflow identities could also be used to authenticate the package publish (restricting publishing to a specific source code repo and removing the need for long-lived access tokens).
+
+**Cons**
+- Limited support today: Only GitHub Actions is fully supported in Fulcio today.
+- Friction: Requires publishing from supported CI/CD providers.
+- CI/CD providers will need to operate an IdP and get it integrated with Sigstore.
+
+Using workload identities along with short-lived X.509 certificates is currently the only solution that meets our key requirements:
```

Comment:
FRSCA, an OpenSSF project, supports this through SPIFEE/Spire: https://github.com/buildsec/frsca -- We could work to integrate a root of trust associated with an OIDC as well.